openedx/openedx-platform
# Add feature to send an e-mail to staff when a student enrolls

This sends an e-mail to a specified e-mail address each time a student enrolls to a course.
- a new lms.env.json variable, `ENROLLMENT_NOTIFICATION_EMAIL`, contains the destination e-mail. Leave it to blank to disable the feature
- a SiteConfiguration variable of the same name and function can be used to enable/disable the feature from admin. If present it will have precedence over the lms.env.json one
- this feature is disabled by default

**JIRA tickets**: None

**Discussions**: None

**Dependencies**: None
Documentation PR not required because this doesn't need a feature flag (`FEATURES` in lms.env.json), but a normal lms.env.json variable.

**Screenshots**:

**Sandbox URL**: TBD - sandbox is being provisioned.

**Partner information**: None

**Deployment targets**: N/A

**Merge deadline**: None

**Testing instructions**:

1. In lms.env.json, set `ENROLLMENT_NOTIFICATION_EMAIL` to your e-mail.
2. Enroll to a course
3. The specified address should receive an e-mail informing of which student enrolled into which course
4. Try to unenroll, no e-mail expected
5. In admin, add a SiteConfiguration object with `{  "ENROLLMENT_NOTIFICATION_EMAIL":"setfromsiteconf@example.com" }`
6. Restart LMS or try the next steps many times, because SiteConfiguration changes sometimes take a short time to be updated
7. Try again the enrollment. You should see the e-mail sent to the new address
8. Try to use '' in SiteConfiguration to disable the feature, while it's enabled in lms.env.json. Retry, it should disable the feature
9. Remove both SiteConfiguration and the lms.env.json setting and retry the enrollment, check that by default it doesn't send e-mail. 

**Author notes and concerns**:

The SiteConfiguration variable allows to do this change from admin without restarting, whereas the lms.env.json one is there because many other e-mail addresses are defined there.

**Reviewers**

**Settings**
```yaml
```
